pkg/sigretests/main.go

```diff
@@ -170,13 +174,49 @@ func getJobsForLatestCommit(org string, repo string, prNumber string) (jobsLastC
 	jobsAllCommits := filterJobs(prHistoryPage)
 	prowjobs = nil
 
-	jobsLatestCommit, err := filterForLastCommit(org, repo, prNumber, latestCommit, jobsAllCommits)
+	jobsLatestCommit, err = filterForLastCommit(org, repo, prNumber, latestCommit, jobsAllCommits)
 	if err != nil {
 		return nil, err
 	}
 	return jobsLatestCommit, nil
 }
 
+func httpGetWithRetry(url string) (resp *http.Response, err error) {
+	httpRetryLog := log.WithField("url", url)
+	var statusCode int
+	retry.Do(
+		func() error {
+			resp, err = http.Get(url)
+			return err
+		},
+		retry.RetryIf(func(err error) bool {
+			httpRetryLog.Warnf("failed to http get (will retry): %v", err)
+			statusCode = httpStatusOrDie(err, httpRetryLog)
+			if statusCode == http.StatusNotFound {
+				return false
+			}
+			if statusCode == http.StatusGatewayTimeout {
+				return true
+			}
+			return false
+		}),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to http get %s: %v", url, err)
+	}
+	return resp, nil
+}
+
+// httpStatusOrDie fetches [stringly typed](https://wiki.c2.com/?StringlyTyped) error code produced by jenkins client
```

Comment:
What is this reference to jenkins client ?

Reply:
oops copy pasta :)

Reply:
i noticed that this whole function is superfluous